build-and-test.ps1

```diff
@@ -0,0 +1,24 @@
+[cmdletbinding()]
+param(
+   [string]$OS="windowsservercore"
+)
+
+Set-StrictMode -Version Latest
+$ErrorActionPreference="Stop"
+
+$dockerRepo="microsoft/dotnet-nightly"
+
+pushd $PSScriptRoot
+
+Get-ChildItem -Recurse -Filter Dockerfile | where DirectoryName -Ilike "*\$OS*" | sort DirectoryName | ForEach {
```

Comment:
Consider using the standard name, `Like`, of the `where` parameter, rather than the alias `ILike`, as I think its meaning is more intuitive.  (When reading this code, at first I thought `ILike` was a variant of `Like` with different functionality.)

Also, consider using lower-case for the `foreach` keyword like for the `where` and `sort` keywords.  Functionally-equivalent alternatives are `ForEach-Object` and `%`.